protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/RcptCmdHandler.java

```diff
@@ -211,9 +211,9 @@ private String getContext(SMTPSession session, MailAddress recipientAddress, Str
         } else if (null != recipient) {
             sb.append(" [to:").append(recipient).append(']');
         }
-        if (null != session.getAttachment(SMTPSession.SENDER, State.Transaction)) {
-            MailAddress mailAddress = (MailAddress) session.getAttachment(SMTPSession.SENDER, State.Transaction);
-            sb.append(" [from:").append(mailAddress.asString()).append(']');
+       MaybeSender sender = (MaybeSender) session.getAttachment(SMTPSession.SENDER, State.Transaction);
+        if (null != sender && !sender.isNullSender()) {
+            sb.append(" [from:").append(sender.get().asString()).append(']');
```

Comment:
why do you need a `.get()` before `asString` ?

Reply:
Very true